packages/shared/lib/core/wallet/utils/getOutputParameters.ts

```diff
@@ -101,35 +108,48 @@ async function buildOutputParametersForLayer2(transactionDetails: NewTransaction
         }
     }
 
-    let estimatedData = await getEstimateData()
-
-    if (estimatedData.gasEstimatePayload.gasBurned) {
-        //  The "+1" is due to an optimization in WASP nodes.
-        const metadata = getLayer2MetadataForTransfer(
-            transactionDetails,
-            (estimatedData.gasEstimatePayload.gasBurned as number) + 1
-        )
-        if (!outputParams.features) {
-            outputParams.features = {}
-        }
-        outputParams.features.metadata = metadata
-        estimatedData = await getEstimateData()
-    }
-
-    // Now that we have the gasFeeCharged, update the amount & the tx details
-    if (estimatedData.gasEstimatePayload.gasFeeCharged) {
-        newTransactionDetails.update((state) => {
-            if (state?.layer2Parameters) {
-                state.layer2Parameters.gasBudget = BigInteger(estimatedData.gasEstimatePayload.gasFeeCharged as number)
+    try {
+        let estimatedData = await getEstimateData()
+
+        if (estimatedData?.gasEstimatePayload?.gasBurned) {
+            //  The "+1" is due to an optimization in WASP nodes.
+            const metadata = getLayer2MetadataForTransfer(
+                transactionDetails,
+                (estimatedData.gasEstimatePayload.gasBurned as number) + 1
+            )
+            if (!outputParams.features) {
+                outputParams.features = {}
             }
-            return state
-        })
-        outputParams.amount = (
-            parseInt(estimatedData.outputForEstimate.amount, 10) + estimatedData.gasEstimatePayload.gasFeeCharged
-        ).toString()
+            outputParams.features.metadata = metadata
+
+            estimatedData = await getEstimateData()
+
+            if (estimatedData?.gasEstimatePayload?.gasFeeCharged) {
+                // Now that we have the gasFeeCharged, update the amount & the tx details
+                newTransactionDetails.update((state) => {
+                    if (state?.layer2Parameters) {
+                        state.layer2Parameters.gasBudget = BigInteger(
+                            estimatedData.gasEstimatePayload.gasFeeCharged as number
+                        )
+                    }
+                    return state
+                })
+                outputParams.amount = (
+                    parseInt(estimatedData.outputForEstimate.amount, 10) +
+                    estimatedData.gasEstimatePayload.gasFeeCharged
+                ).toString()
+
+                return outputParams
+            }
+        }
+    } catch (error) {
+        if (error?.message) {
+            showAppNotification({
+                type: 'error',
+                message: error?.message,
+            })
+        }
```

Comment:
Is it really necessary to catch the error here? I think you can just let it bubble up and it will be catched in `prepareTransactionOutput()`